packages/@react-aria/interactions/src/useFocusVisible.ts

```diff
@@ -201,7 +201,8 @@ export function useInteractionModality(): Modality | null {
  * focus visible style can be properly set.
  */
 function isKeyboardFocusEvent(isTextInput: boolean, modality: Modality, e: HandlerEvent) {
-  return !(isTextInput && modality === 'keyboard' && e instanceof KeyboardEvent && !FOCUS_VISIBLE_INPUT_KEYS[e.key]);
+  let elementType = (e?.target as HTMLInputElement)?.type;
+  return !((isTextInput || elementType === 'text' || elementType === 'textarea') && modality === 'keyboard' && e instanceof KeyboardEvent && !FOCUS_VISIBLE_INPUT_KEYS[e.key]);
```

Comment:
wouldn't textarea be the tagName rather than the `type` attribute? There are also other input types that accept text input, like number, password, etc. I think we basically want the `willOpenKeyboard` function and input type list from `usePreventScroll`? I guess I'd just copy it here.